### ISSUE TITLE
Feature: LMS Dynamic navigation

### ISSUE THREAD

**Issue by Fermain** [Fermain/-mollify#23]
LMS requires a navigation module capable of dynamically rendering menus based on frontmatter information. For example, a course with the following frontmatter:

```yml
title: Example Course
slug: example-course
tags:
- example
- course
children:
- my-first-module
- my-second-module
- course evaluation
```

Has the following Navigation:

```md
## [Example Course](.)

- [My First Module](my-first-module)
- [My Second Module](my-second-module)
- [Course Evaluation](course-evaluation)
```

**Suggested approach:**
1. Clone the `@mollify/lms` package
2. Create a branch for navigation
3. Scaffold a hierarchy of test files using the format below
4. Create a `Navigation` component within `src/lib/components`
5. Create an MVP approach using a static list of pages (see below)
6. Bring your findings to the team to discuss next steps

A full implementation will require additional CLI tools to build an index of a folder which is why it is suggested to proceed with sample data for now.

```
/src/routes/example-course/+page.md
/src/routes/example-course/example-module-1/+page.md
/src/routes/example-course/example-module-2/+page.md
/src/routes/example-course/example-module-1/example-lesson-1/+page.md
/src/routes/example-course/example-module-1/example-lesson-2/+page.md
```

```md
---
title: Example Course
slug: example-course
children:
- example-module-1
- example-module-2
---
```

```md
---
title: Example Module
slug: example-module
children:
- example-lesson-1
- example-lesson-2
---
```

**Comment by Fermain** [comment]
After a team discussion it was decided that the navigation should walk the file tree instead of using metadata. This simplifies things but creates a new issue - menu ordering.

Since there are plans to implement dependencies, these could be used to form the menu order. For example:

```
title: Introduction to X
dependencies:
```

```
title: Getting Started with X
dependencies:
- Introduction to X
```

```
title: Mastering X
dependencies:
- Getting Started with X
```

This would produce this menu:
- [Introduction to X](#)
- [Getting Started with X](#)
- [Mastering X](#)

**Comment by Fermain** [comment]
Related to #24

**Comment by Anclagen** [comment]
A basic version of content navigation is now implemented and ready for merge #31 . The file structure is mapped out to a nesting of object, with frontmatter being grabbed for all the markdown files. This data is then filtered further to get the current pages specific content.

Currently 3 navigation components have been added. 
- A recursive navigation that filters the entire file structure object and outputs navigation for everything.
- A institute and programme navigation that appends a grid of cards to the main
- A course navigation that shows the current course your viewing with all its modules and lessons.

Further additions can be added through sorting based on dependencies #24 task and possibly with considerations for #30 as the required structures might overlap.
